zbus/src/connection/connect/mod.rs

```diff
@@ -0,0 +1,62 @@
+use std::{future::Future, pin::Pin};
+use tracing::debug;
+
+use crate::{
+    address::{transport::Transport, Address},
+    Error, Guid, OwnedGuid, Result,
+};
+
+use super::socket::{self, BoxedSplit};
+
+mod macos;
+mod win32;
+
+type ConnectResult = Result<(BoxedSplit, Option<OwnedGuid>)>;
```

Comment:
Could you please go through other code added here and change it to follow the top-down layout (something I've been trying my best to do for a few years now). I handled this file already here: https://github.com/dbus2/zbus/pull/1036/commits/cd799e4790dc987a1912c93a3b99275a67b045f6

Reply:
I can't really think of a better order, please do it as it suits you.

Reply:
> I can't really think of a better order,

Not sure what's there to think of. It's like this:

* pub > pub(crate) > private.
* Use before declaration/definition.

Think in terms of what you most likely want to see first in a file when you open it? Most likely it's the public API or pub(crate) API. Also, you want to see the high-level code (i-e use) before implementation details.

>  please do it as it suits you.

I'm sorry but you've to follow the conventions of the project. Besides, I did some of the work for you and gave you an example to follow.

Reply:
Having said all that, this is not super important/urgent. But please keep it in mind in the future. I wish I could say the same about the `address::Error` discussion above but currently it's neither my way nor your way.

Reply:
done in #1050, if you want further code ordering, please be specific, or I invite you to do so. thanks